### PR TITLE
0.14.2-incubating download links, docs

### PR DESCRIPTION
**Pull request by clintropolis**
Adds download link for 0.14.2-incubating, slightly reworks download page to serve only latest releases from the latest 2 release branches the mirrors, and adds a link to `archive.apache.org` at the bottom of the downloads page in a 'Older releases' section.

Screen Shot 2019-05-28 at 4 00 23 PM

...

Screen Shot 2019-05-28 at 4 00 32 PM

in `_config.yml` the versions property has been modified to group versions by release:
```
druid_versions:
  - release: 0.14
    versions:
      - version: 0.14.2-incubating
        date: 2019-05-26
      - version: 0.14.1-incubating
        date: 2019-05-09
      - version: 0.14.0-incubating
        date: 2019-04-09
  - release: 0.13
    versions:
      - version: 0.13.0-incubating
        date: 2018-12-18
```
which the downloads page takes the first version of the first 2 groups to link to mirrors. News sidebar is essentially unchanged.

Screen Shot 2019-05-26 at 3 34 26 PM

### CODE REVIEW

**Comment by fjy** [comment]
@clintropolis please keep links to older versions available

**Comment by fjy** [comment]
@clintropolis did the older releases get moved to the bottom of the page?

**Comment by clintropolis** [comment]
> @clintropolis did the older releases get moved to the bottom of the page?

Yeah, I moved them to the bottom to downplay the older releases. The are going to be served from the archives per apache policy that only latest releases be on the mirrors, which doesn't support the `/dyn/closer.cgi?path=` stuff that the mirrors do which I think plays into how downloads are tracked.

**Comment by fjy** [comment]
I would prefer this not merged until we understand impacts to downloads

**Comment by clintropolis** [comment]
>I would prefer this not merged until we understand impacts of all changes in webpage

Anything I can do to help? This is the last thing I have to do before I can officially announce the release.

Which part are you worried about, moving the older releases to the bottom of the page or the older releases being served from the archive?

Not sure there is much I can do about the latter, but download pages for other Apache projects are similar where only actively developed branches are served from the mirrors. [Taking Apache Kafka for example](https://kafka.apache.org/downloads), the latest 2.2.x and 2.1.x releases are from the mirrors, the rest are from the archives. [Apache Spark](https://spark.apache.org/downloads.html) only has the latest release available from the mirrors on their download page, and a single link to the archives for all older releases. [Apache Hadoop](https://hadoop.apache.org/releases.html) has several active release lines so has all of those from the mirrors on the download page, and like Spark a link to the archives for any release not on the page.

We would be ok with 0.14.2-incubating and 0.13.0-incubating being on the mirrors since they are the latest of 0.14.x and 0.13.x, and 0.14.1 and 0.14.0 in the archives, but since we aren't actively developing for 0.13.0-incubating I thought it made more sense for it to live chronologically in the 'Older releases' section instead.

**Comment by gianm** [comment]
> I would prefer this not merged until we understand impacts of all changes in webpage

@fjy could you elaborate on what potential impacts you think we should investigate?

@clintropolis @fjy how do you feel about keeping the set of releases on the mirrors and on downloads.html matching (ie avoid archive links)? Post 0.14.2 release it should probably be either just 0.14.2, or 0.14.2 plus 0.13.0. (i.e. latest patch release from the last two major lines)

**Comment by clintropolis** [comment]
>@clintropolis @fjy how do you feel about keeping the set of releases on the mirrors and on downloads.html matching (ie avoid archive links)? Post 0.14.2 release it should probably be either just 0.14.2, or 0.14.2 plus 0.13.0. (i.e. latest patch release from the last two major lines)

That sounds fine to me, I don't have very strong opinions on this. I think in that case I would put both 0.14.2 and 0.13.0 in the top again like the current layout of the page, and then at the bottom instead of enumerating all the archive releases, change the heading to 'Release archive' and link to  https://archive.apache.org/dist/incubator/druid/

**Comment by clintropolis** [comment]
Likewise, probably the same should be done in apache site here https://github.com/apache/incubator-druid-website/pull/6 if we go with this approach, with 0.14.2 an 0.13.0 in the table and a link below to the archives

**Comment by clintropolis** [comment]
I have updated the PR description based on discussion so far

**Comment by fjy** [comment]
👍 if we're certain it won't negatively impact webpage

File: downloads.md [hunk 1]
```diff
@@ -59,3 +56,7 @@ Release candidates are announced on the [Druid dev mailing list](https://lists.a
 If you are interested in being notified of upcoming release candidates and being involved in the dev community (e.g. validating and approving RC builds), you can subscribe to the list by emailing: **dev-subscribe@druid.apache.org**
 
 Previous discussions are available in the [Apache Mail Archives](https://lists.apache.org/list.html?dev@druid.apache.org).
+
+## Release archive
+
+All previous Apache releases of Druid are available at the [Apache release archives](https://archive.apache.org/dist/incubator/druid/).
```
[the remark this answers is not in the document]
Comment (gianm) [reply downloads.md]: Oops, I forgot to remove this comment.

**Pull request merged**